### ISSUE TITLE
Format markdown cells with rmarkdown v2

### ISSUE THREAD

**Issue by gaborcsardi** [att/rcloud#2284]
Instead of the current (knitr + markdown package which is sundown) way.

This is needed to support proper HTML widgets in markdown cells, see https://github.com/att/rcloud.htmlwidgets/issues/1

**Comment by gordonwoodhull** [comment]
I agree that in the long run we want to migrate from knitr to Markdown. I'm not sure if there is anything our Markdown cells do that our RMarkdown cells don't. Markdown is probably a little faster because there is less disk access and no external processes. 

It looks like htmlwidgets currently work in RMarkdown cells (with the `suppress_viewer` hack!?) but this is probably by accident and looks like it's out of order - like maybe we're still intercepting print and diverting the output. 

The RMarkdown support is currently "experimental" in the sense that I wrote it for a particular purpose (Mani's rcloud.iip slideshow package), never really tested it to find out what works and what doesn't, but it seemed to work, so we shipped it and and never looked back.

As we discussed on the call, the real prize here would be to process the intermediate htmltools structures similar to what we're doing for htmlwidgets. If that makes sense, it could be a lot more efficient and we should be able to fix RMarkdown features that aren't working.

**Comment by gaborcsardi** [comment]
First of all, I realized, that we do not need html widget support in markdown/rmarkdown cells for flexdashboard. It is enough to have html widgets support in regular R cells, and we already have that. So I think we can close this issue.

> It looks like htmlwidgets currently work in RMarkdown cells (only with the suppress_viewer hack!?) but this is probably by accident and looks like it's out of order - like maybe we're still intercepting print and diverting the output.

That is indeed surprising. They did not work for me, I only got a static image (with the wrong size), but maybe I did not have pandoc on the rcloud machine.

Making rmarkdown cells work with the optimized html widgets code seems really hard to me, so how about I close this issue now and we focus on flexdashboard?

**Comment by gordonwoodhull** [comment]
I'd rather keep it open for the future, but sure, it's not something you have to worry about.

**Comment by gaborcsardi** [comment]
OK, sorry. :)

**Comment by gordonwoodhull** [comment]
NP. It does seem like the right way to go, for a variety of reasons.

I guess different projects have different ways of managing issues. Because RCloud has such a ridiculous scope we're used to having issues open for years into the future. (One of our milestones used to be called "when hell freezes over" but now it's called "when things get slow", which is equivalent. ;-)

I'll put this in "2.0".
